File: ecobertura.core/src/ecobertura/core/log/LogStatus.java
package ecobertura.core.log;

import org.eclipse.core.runtime.Status;

import ecobertura.core.CorePlugin;

public class LogStatus extends Status {
	public static LogStatus fromMessageWithSeverity(final String message, final int severity) {
		assert message != null;
		assert severity == Status.ERROR || severity == Status.WARNING || severity == Status.INFO;
		
		return new LogStatus(severity, message);
	}
	
	public static LogStatus fromExceptionWithSeverity(final String message, final Throwable throwable, 
			final int severity) {
		assert message != null;
		assert severity == Status.ERROR || severity == Status.WARNING || severity == Status.INFO;
		assert throwable != null;
		
		return new LogStatus(severity, message, throwable);
	}
	
	private LogStatus(final int severity, final String message) {
		super(severity, CorePlugin.PLUGIN_ID, message);
	}
	
	private LogStatus(final int severity, final String message, final Throwable throwable) {
		super(severity, CorePlugin.PLUGIN_ID, message, throwable);
	}
	
}

File: ecobertura.core/src/main/scala/ecobertura/core/log/LogStatus.scala
/*
 * This file is part of eCobertura.
 * 
 * Copyright (c) 2009, 2010 Joachim Hofer
 * All rights reserved.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
package ecobertura.core.log

import org.eclipse.core.runtime.IStatus
import org.eclipse.core.runtime.Status

import ecobertura.core.CorePlugin

object LogStatus {
  
  object Severity extends Enumeration {
    val Error = Value(IStatus.ERROR)
    val Warning = Value(IStatus.WARNING)
    val Info = Value(IStatus.INFO)
  }
      
  def something = Severity.Info
  
  def fromMessageWithSeverity(message: String, severity: Severity.Value) = { 
    require(message != null)
    new Status(severity.id, CorePlugin.pluginId, message)
  }
  
  def fromExceptionWithSeverity(message: String, throwable: Throwable, severity: Severity.Value) = {
    require(message != null)
    require(throwable != null)
    new Status(severity.id, CorePlugin.pluginId, message, throwable)    
  }
}